test/AppNotificationBuilderTests/APITests.cpp

```diff
@@ -382,5 +382,118 @@ namespace Test::AppNotification::Builder
+
+        TEST_METHOD(AppNotificationBuilderAddTooManyInputElements)
+        {
+            VERIFY_THROWS_HR(AppNotificationBuilder()
+                .AddComboBox(AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox2")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox3")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox4")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox5")
+                    .AddItem(L"item1", L"item1 text"))
+                .AddComboBox(AppNotificationComboBox(L"comboBox6")
+                    .AddItem(L"item1", L"item1 text")), E_INVALIDARG);
+        }
+
+        TEST_METHOD(AppNotificationComboBoxAddTooManySelectionItems)
+        {
+            VERIFY_THROWS_HR(AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text")
+                    .AddItem(L"item2", L"item2 text")
+                    .AddItem(L"item3", L"item3 text")
+                    .AddItem(L"item4", L"item4 text")
+                    .AddItem(L"item5", L"item5 text")
+                    .AddItem(L"item6", L"item6 text"), E_INVALIDARG);
+        }
+
+        TEST_METHOD(AppNotificationComboBoxAddFiveSelectionItems)
+        {
+            auto comboBox{ AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text")
+                    .AddItem(L"item2", L"item2 text")
+                    .AddItem(L"item3", L"item3 text")
+                    .AddItem(L"item4", L"item4 text")
+                    .AddItem(L"item5", L"item5 text") };
+            auto expected{ L"<input id='comboBox1' type='selection'><selection id='item1' content='item1 text'/><selection id='item2' content='item2 text'/><selection id='item3' content='item3 text'/><selection id='item4' content='item4 text'/><selection id='item5' content='item5 text'/></input>" };
+
+            VERIFY_ARE_EQUAL(comboBox.as<winrt::Windows::Foundation::IStringable>().ToString(), expected);
+        }
+
+        TEST_METHOD(AppNotificationComboBoxAddSelectionItemWithoutAnId)
+        {
+            VERIFY_THROWS_HR(AppNotificationComboBox(L"comboBox1")
+                .AddItem(L"", L"item text"), E_INVALIDARG);
+        }
+
+        TEST_METHOD(AppNotificationComboBoxAddTwoSelectionItemsWithSameId)
+        {
+            auto comboBox{ AppNotificationComboBox(L"comboBox1")
+                    .AddItem(L"item1", L"item1 text")
+                    .AddItem(L"item1", L"item2 text") };
+            auto expected{ L"<input id='comboBox1' type='selection'><selection id='item1' content='item2 text'/></input>" };
+
+            VERIFY_ARE_EQUAL(comboBox.as<winrt::Windows::Foundation::IStringable>().ToString(), expected);
```

Comment:
Recommend using builder.BuildNotification to test developer flow.

Reply:
Notr everything needs to go through the AppNotificationBuilder. These are unit tests, and the smaller they are, the easier it is to maintain and debug them.

Some already go through the AppNotificationBuilder to validate that the content gets inserted correctly. Others are just verifying the class works as per the specs.